server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -296,10 +322,11 @@ void innerDetect(
             .append(", ignoreIdleThreads=")
             .append(Boolean.toString(ignoreIdleThreads))
             .append(":\n");
+        onNextThread.run();
 
         // Capture before and after thread state with timings
         Map<Long, ThreadTimeAccumulator> previousThreadInfos = getAllValidThreadInfos(threadBean, sunThreadInfo, currentThreadId);
-        threadSleep.apply(interval.millis());
+        Thread.sleep(interval.millis());
```

Comment:
This slows the tests down doesn't it? Previously we were not actually sleeping here at all, whereas now we do. Is this change needed for this utility?